airbyte-webapp/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,234 @@
+import classNames from "classnames";
+import { Line } from "rc-progress";
+import { useState } from "react";
+import { useIntl } from "react-intl";
+
+import { getJobStatus } from "components/JobItem/JobItem";
+
+import { AttemptRead, JobConfigType, SynchronousJobRead } from "core/request/AirbyteClient";
+import Status from "core/statuses";
+import { JobsWithJobs } from "pages/ConnectionPage/pages/ConnectionItemPage/JobsList";
+import { formatBytes } from "utils/numberHelper";
+
+import styles from "./ProgressBar.module.scss";
+
+function isJobsWithJobs(job: JobsWithJobs | SynchronousJobRead): job is JobsWithJobs {
+  return (job as JobsWithJobs).attempts !== undefined;
+}
+
+const buttonUnStyle = {
+  background: "none",
+  color: "inherit",
+  border: "none",
+  padding: 0,
+  font: "inherit",
+  cursor: "pointer",
+  outline: "inherit",
+  textDecoration: "underline",
+};
+
+export const ProgressBar = ({
+  job,
+  jobConfigType,
+}: {
+  job: JobsWithJobs | SynchronousJobRead;
+  jobConfigType: JobConfigType;
+}) => {
```

Comment:
We're extracting the property type always in it's own interface named `ProgressBarProps`, also `ProgressBar` should be explicitally types as `React.FC<ProgressBarProps>` to catch errors of wrong return types within this file instead of the consuming JSX place.